Source/ASNodeController+Beta.m

```diff
@@ -89,3 +108,41 @@ - (void)interfaceStateDidChange:(ASInterfaceState)newState
                       fromState:(ASInterfaceState)oldState {}
 
 @end
+
+@implementation ASDisplayNode (ASNodeControllerOwnership)
+
+- (ASNodeController *)__nodeController
+{
+  ASNodeController *nodeController = nil;
+  id object = objc_getAssociatedObject(self, @selector(__nodeController));
+
+  if ([object isKindOfClass:[ASWeakProxy class]]) {
+    nodeController = (ASNodeController *)[(ASWeakProxy *)object target];
+  } else {
+    nodeController = (ASNodeController *)object;
+  }
+
+  return nodeController;
+}
+
+- (void)__setNodeControllerStrong:(ASNodeController *)nodeController
+{
+  objc_setAssociatedObject(self, @selector(__nodeController), nodeController, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+
+- (void)__setNodeControllerWeak:(ASNodeController *)nodeController
+{
+  // Associated objects don't support weak references. Since assign can become a dangling pointer, use ASWeakProxy.
+  ASWeakProxy *nodeControllerProxy = [ASWeakProxy weakProxyWithTarget:nodeController];
+  objc_setAssociatedObject(self, @selector(__nodeController), nodeControllerProxy, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
```

Comment:
I'm cool with this, but why not add 2 `@package` ivars to ASDisplayNode and use those instead?